src/components/molecules/bookCard/BookCard.tsx

```diff
@@ -1,29 +1,65 @@
-import React from 'react';
-import Card from '@mui/material/Card';
-import BookImage from '../../atoms/bookImage/BookImage';
-import BookDetails from '../bookDetails/BookDetails';
-import Box from '@mui/material/Box';
+import React, { useState } from "react";
+import Card from "@mui/material/Card";
+import BookImage from "../../atoms/bookImage/BookImage";
+import BookDetails from "../bookDetails/BookDetails";
+import { ButtonProps } from "@mui/material";
+import Box from "@mui/material/Box";
 
-type Props = {
-    coverImage:string;
-    bookTitle:string;
-    bookAuthor:string;
-    time:string;
-};
+interface Props extends ButtonProps {
+  coverImage: string;
+  bookTitle: string;
+  bookAuthor: string;
+  time: string;
+  isButtoned: Boolean;
+  btnText?: string;
+  handleClick?: any;
+  index: number;
+}
 
 function BookCard(props: Props) {
-    const{coverImage,bookTitle,bookAuthor,time} = props;
-  return <div>
-      <Card sx={{
-          maxWidth:'284px',
-          backgroundColor:'formBg.main',
-          borderRadius:'8px',
-          pb:4
-      }}>
-          <BookImage source={coverImage}/>
-          <BookDetails bookTitle={bookTitle} bookAuthor={bookAuthor} time={time}/>
+  const {
+    coverImage,
+    bookTitle,
+    bookAuthor,
+    time,
+    isButtoned,
+    btnText,
+    startIcon,
+    index,
+    handleClick,
+  } = props;
+  let paddingBottom = 0;
+  const [isHovered, setHovered] = useState(false);
+  if (!isButtoned) {
+    paddingBottom = 4;
+  }
+  
+  return (
+    <div>
+      <Card
+        sx={{
+          maxWidth: "284px",
+          backgroundColor: "formBg.main",
+          borderRadius: "8px",
+          pb: paddingBottom,
```

Comment:
Instead of creating a new variable paddingBottom you can directly write pb: !isButtoned ? 4 : 0 here